### PR TITLE
dunify and make the initialization parallel

### PR DESCRIPTION
**Pull request by totto82**
Tested on norne and it gives the same initial values in parallel and in serial. Should be tested on more models.

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
The test failures on MOD4_GRP is expected since this PR uses the average of the four corners of the top/bottom cell and not the centroid of the face to determine the cells vertical span. I can go back to using the face centroid, but I think the average of the corners is more aligned with how we compute cell centers etc in the code. Any opinion?

**Comment by totto82** [comment]
When I compare the initial results in resInsight they are exactly the same. I will rerun jenkins to make sure I don't run after ghosts.

**Comment by totto82** [comment]
jenkins build this please

File: ebos/eclequilinitializer.hh [hunk 1]
```diff
@@ -100,7 +100,7 @@ public:
 
         EQUIL::DeckDependent::InitialStateComputer<TypeTag> initialState(materialLawManager,
                                                                          eclState,
-                                                                         vanguard.grid(),
+                                                                         vanguard,
```
Comment (blattms) [inline ebos/eclequilinitializer.hh:103]: I would appreciate it if we would not pass the vanguard here, but the gridView. I seems like we only need to access the view itseld, the grid (accessible via `gridView.grid()` and the communicator (accessible via `gridView.grid().comm()`.

**Review by blattms — APPROVED** [review]
I generally approve of this but would prefer to not use the vanguard, if possible,

File: ebos/eclequilinitializer.hh (hunk 1 above)
Comment (totto82) [reply ebos/eclequilinitializer.hh:103]: I agree. I will update.

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
The initial results of the test cases that fails are the same. i.e. the difference = 0 for all cells when compared in ResInsight. When comparing the simulations I notice a difference in number of newton iterations after a few time step that leads to slight change of timesteps that further leads to difference in when the simulator has problems to converge. A visual inspection of the results shows agreement. IMO it is safe to do a update_data and merge this.

**Comment by totto82** [comment]
@atgeirr @bska Any comments? Great if we can get this in soon.

**Review by bska — COMMENTED** [review]
Is there a reason you're copying all the `Geometry` objects?  Also, `inline` is superfluous for templates.  Those are implicitly inline.

**Comment by totto82** [comment]
> Is there a reason you're copying all the Geometry objects? Also, inline is superfluous for templates. Those are implicitly inline.

Thanks for noticing. I have pushed a fix.

**Comment by totto82** [comment]
jenkins build this please

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
jenkins build this please

**Comment by totto82** [comment]
This needs an update data. I have verified that the test failures are only numerical noise.

**Review by bska — APPROVED** [review]
Looks good to me.  Note that the requisite update to the reference solutions is going to conflict with the existing update for #2949 (i.e., OPM/opm-tests#455).  One of these has to go in first and then the other will likely have to be redone.

**Comment by totto82** [comment]
Lets get #2949 in first. I will fix the conflict and update

**Comment by totto82** [comment]
@bska Are https://github.com/OPM/opm-common/pull/2119 and https://github.com/OPM/opm-common/pull/2164 ready for merging? Or can I try to get this in first?

**Comment by bska** [comment]
> can I try to get this in first?

I think we should merge this PR first.  It's been ready for a while and the other two aren't approved yet.  I'll generate updated reference solutions.

**Comment by bska** [comment]
jenkins build this update_data please

**Comment by bska** [comment]
jenkins build this opm-tests=458 please

**Comment by bska** [comment]
New reference data is installed on the CI system.  I'm merging this now to bring the system back to a consistent state.  Thanks a lot for the new feature and for your patience.

File: ebos/equil/initstateequil.hh [hunk 2]
```diff
@@ -1773,6 +1770,28 @@ private:
...
+        cellCenterDepth_.resize(numElements);
+        cellZSpan_.resize(numElements);
+        cellZMinMax_.resize(numElements);
+
+        auto elemIt = gridView.template begin</*codim=*/0>();
+        const auto& elemEndIt = gridView.template end</*codim=*/0>();
+        for (; elemIt != elemEndIt; ++elemIt) {
+            const Element& element = *elemIt;
+            const unsigned int elemIdx = elemMapper.index(element);
+            cellCenterDepth_[elemIdx] = Details::cellCenterDepth(element);
```
Comment (GitPaean) [inline ebos/equil/initstateequil.hh:1790]: One question here, why not use the `cellCenterDepth_` from the `vanguard` here? Thanks. @totto82

Comment (GitPaean) [reply ebos/equil/initstateequil.hh:1790]: Never mind. I thought without this can simply some work for me. It is not case.

**Pull request merged**